Clarify how rule timeout works (#1960)
semgrep/semgrep-docs: docs/deployment/customize-ci-jobs.md

## Set a custom timeout

> By default, Semgrep spends **5 seconds** to run **per rule** on each relevant file. To **set a different timeout**, set the `SEMGREP_TIMEOUT` environment variable (the value is in seconds). Decreasing this value speeds up your scans, but with the possibility of skipping some rules. Alternatively, increasing this value ensures that your most complex rules finish running. For example:

Reviewer: ```suggestion
By default, Semgrep spends **5 seconds** to run **per rule** per file scanned. To **set a different timeout**, set the `SEMGREP_TIMEOUT` environment variable (the value is in seconds). Decreasing this value speeds up your scans, but with the possibility of skipping some rules. Alternatively, increasing this value ensures that your most complex rules finish running. For example:
```

Author: @khorne3 I'm definitely open to revising 'relevant' to something more clear, but this is not clear, unfortunately. It doesn't clarify that every rule does not scan every file but only files which are within the language targets. I'm pretty sure we do talk about this elsewhere but it's not easy to search for so I don't know exactly where. If there's a good / approved way to talk about it I can use that here too.

Reviewer: Got it -- maybe "relevant file scanned"? It seems like both pieces are crucial to understanding

Author: Yeah, I think I actually didn't understand the purpose of your change either, sorry. If relevant is good enough as a word, we can just do both, but let me see if I can come up with something both short and clear.

Author: I updated, mind taking another look?
